### PR TITLE
Enable arg file for resolver path to avoid arg too long errors.

### PR DESCRIPTION
**Pull request by karthikbalasub**

### CODE REVIEW

File: generator/src/main/java/com/linkedin/pegasus/generator/PegasusDataTemplateGenerator.java [hunk 1]
```diff
@@ -97,7 +98,15 @@ public static void main(String[] args)
 
     final String generateImportedProperty = System.getProperty(PegasusDataTemplateGenerator.GENERATOR_GENERATE_IMPORTED);
     final boolean generateImported = generateImportedProperty == null ? true : Boolean.parseBoolean(generateImportedProperty);
-    PegasusDataTemplateGenerator.run(System.getProperty(AbstractGenerator.GENERATOR_RESOLVER_PATH),
+    String resolverPath = System.getProperty(AbstractGenerator.GENERATOR_RESOLVER_PATH);
+    if (resolverPath != null && ArgumentFileProcessor.isArgFile(resolverPath))
```
Comment (DPUkyle) [inline generator/src/main/java/com/linkedin/pegasus/generator/PegasusDataTemplateGenerator.java:102]: Nice. Thanks @evanw555 for convincing me to extract `isArgFile(String)` to a utility class.

File: gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java [hunk 2]
```diff
@@ -1493,6 +1493,7 @@ protected void configureConversionUtilities(Project project, SourceSet sourceSet
       }
       task.setKeepOriginal(keepOriginal);
       task.setSkipVerification(skipVerification);
+      task.setEnableArgFile(true);
```
Comment (DPUkyle) [inline gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java:1496]: Are you sure you want to globally enable this feature? Evan and I reserved it as off by default, and allowing consumers with javaexec issues to opt-in in their build.gradle files.

Comment (evanw555) [reply gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java:1496]: +1 on this. We've seen before that `gradle-plugins` and `restli-tools` are not always locked to the same version. Kyle and I think it's better to keep it opt-in for now, then make it enabled by default later (or if/when we create newer more resilient tools)

Comment (karthikbalasub) [reply gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java]: Done

Comment (evanw555) [reply gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java]: Actually, after thinking about this a little harder, I've realized that: `convertXXXToPdl` and `reformatXXXPdl` are not part of the normal build process, so enabling arg file by default may not be that risky. However, the same cannot be said for `generateXXXDataTemplate` or `XXXTranslateSchemas`; these should be opt-in IMO.

File: gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java [hunk 3]
```diff
@@ -527,6 +527,7 @@
   // by default it is disabled, because it triggers hot-reload (even if it results in a no-op)
   private static final String DESTROY_STALE_FILES_ENABLE = "enableDestroyStaleFiles";
   public static final Collection<String> DATA_TEMPLATE_FILE_SUFFIXES = new ArrayList<>();
+  private static final String DATA_TEMPLATE_ENABLE_ARGFILE = "generateDataTemplate.enableArgFile";
```
Comment (evanw555) [inline gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java:530]: Is this necessary? Task inputs can be configured directly as such:
```
generateMainDataTemplate {
  enableArgFile = true
}
```
This is how the fix for `generateXXXRestClient` is enabled.

File: gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java [hunk 4]
```diff
@@ -546,6 +547,7 @@
   private static final String CONVERT_TO_PDL_KEEP_ORIGINAL = "convertToPdl.keepOriginal";
   private static final String CONVERT_TO_PDL_SKIP_VERIFICATION = "convertToPdl.skipVerification";
   private static final String CONVERT_TO_PDL_PRESERVE_SOURCE_CMD = "convertToPdl.preserveSourceCmd";
+  private static final String TRANSLATE_SCHMEAS_ENABLE_ARGFILE = "translateSchemas.enableArgFile";
```
Comment (evanw555) [inline gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java:550]: nit: misspelled "schemas"

File: gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java (hunk 3 above)
Comment (karthikbalasub) [reply gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java]: Remove the property based logic. We can enable this using task args directly.

**Pull request merged**